app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,221 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from rest_framework.test import APIClient
+from rest_framework import status
+
+from recipe.models import (
+    Recipe,
+    Ingredient,
+)
+from recipe.serializers import RecipeSerializer
+
+
+RECIPES_URL = reverse('recipe:recipe-list')
+
+
+def detail_url(recipe_id):
+    return reverse('recipe:recipe-detail', args=[recipe_id])
+
+
+def create_recipe(**params):
+    defaults = {
+        'name': 'Sample recipe name',
+        'description': 'Sample recipe description',
+    }
+    defaults.update(params)
+
+    recipe = Recipe.objects.create(**defaults)
+    return recipe
+
+
+class RecipeApiTest(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_get_recipes(self):
+        create_recipe(name='Breakfast')
+        create_recipe(name='Lunch')
+
+        res = self.client.get(RECIPES_URL)
+
+        recipes = Recipe.objects.all().order_by('-id')
+
+        serializer = RecipeSerializer(recipes, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_get_recipe_detail(self):
+        recipe = create_recipe(name='Dinner')
+
+        url = detail_url(recipe.id)
+        res = self.client.get(url)
+
+        serializer = RecipeSerializer(recipe)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_recipe(self):
```

Comment:
This test is called "test_create_recipe" so the main thing to test here is that if we call the API with a valid payload it will create a Recipe object on our DB. You are checking that the return of your API serialized is the same as what we have on the DB and this is a bit misleading because maybe the API returns a DTO with not all the info. The best way is to check with your initial values in the payload so you don't need even the serializer.

Reply:
Yeah right, I have replaced comparison using serializer with object data comparison from DB.